tango_with_django_project/rango/views.py

```diff
@@ -244,6 +268,39 @@ def user_login(request):
         return render(request, 'rango/login.html')
 
 
+class AddReviewView(View):
+
+    @method_decorator(login_required)
+    def get(self, request, slug):
+        form = ReviewForm()
+        product = Product.objects.get(slug=slug)
+
+        if product is None:
+            return redirect(reverse('rango:index'))
+
+        context_dict = {'form': form, 'product': product}
+        return render(request, 'rango/review.html', context=context_dict)
+
+    def post(self, request, slug):
+        form = ReviewForm(request.POST)
+        product = Product.objects.get(slug=slug)
+
+        if product is None:
+            return redirect(reverse('rango:index'))
+        if form.is_valid():
+            if product:
+                review = form.save(commit=False)
+                review.product = product
+                review.user = request.user
+                review.save()
+                return redirect(reverse('rango:index'))
```

Comment:
Just my idea; Wouldn't it be better to redirect to the previous page so user can check the reviews without going thru all the way from `index` to `review`? 

Other than this, I'm so happy with all the changes